projects/plugins/jetpack/extensions/blocks/blogroll/utils.js

```diff
@@ -20,3 +20,22 @@ export function createBlockFromSubscription( subscription ) {
 		description,
 	} );
 }
+
+export function checkIfValidDomain( siteURL ) {
+	const regEx =
+		/(http(s)?:\/\/.)?(www\.)?[-a-zA-Z0-9@:%._+~#=]{2,256}\.[a-z]{2,64}\b([-a-zA-Z0-9@:%_+.~#?&//=]*)/g;
+
+	if ( ! siteURL ) {
+		return false;
+	}
+
+	let validUrl;
+
+	try {
+		validUrl = new URL( siteURL );
+	} catch ( e ) {
+		validUrl = siteURL.match( regEx );
```

Comment:
We should use the match, not the input.
If somebody gives http://google.com/hello, we only want the google.com.
```suggestion
		return new URL( siteURL ).host;
	} catch ( e ) {
		return siteURL.match ( regEx );
```

Reply:
@alshakero - I updated the logic to return the valid domain or null. It will either get the host name from the new URL constructor or attempt to get the host from the regex match.